plan/views.py

```diff
@@ -32,3 +39,83 @@ def get_queryset(self):
 
 class CourseDetailSearch(CourseDetail):
     serializer_class = CourseDetailWithReviewSerializer
+
+
+def get_sections(data):
+    sections = None
+    if 'meetings' in data:
+        sections = []
+        for s in data.get('meetings'):
+            _, section = get_course_and_section(s.get('id'), s.get('semester'))
+            sections.append(section)
+    elif 'sections' in data:
+        sections = []
+        for s in data.get('sections'):
+            _, section = get_course_and_section(s.get('id'), s.get('semester'))
+            sections.append(section)
+    return sections
+
+
+class ScheduleViewSet(viewsets.ModelViewSet):
+    serializer_class = ScheduleSerializer
+    http_method_names = ['get', 'post', 'delete', 'put']
+    permission_classes = [IsAuthenticated]
+
+    def update(self, request, pk=None):
+        existing_obs = Schedule.objects.filter(id=pk)
+        if (len(existing_obs) == 0):
+            return Response({'detail': 'Not found.'}, status=status.HTTP_404_NOT_FOUND)
+
+        if 'semester' not in request.data:
+            request.data['semester'] = get_value('SEMESTER', None)
+
+        for s in request.data['sections']:
+            if s.get('semester') != request.data.get('semester'):
+                return Response({'detail': 'Semester uniformity invariant violated.'},
+                                status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            existing_obs.update(person=request.user,
```

Comment:
`existing_obs` and `ob` are pretty undescriptive variable names, especially because they refer to almost the same thing?